### PR TITLE
Added page titles all over the MM app

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
fixes https://github.com/mitodl/micromasters/issues/449

#### What's this PR do?
- Adds page title all over the application.

#### How should this be manually tested?
Go to each page in app and see the tab title.

@pdpinch 
#### screenshot
screen shot 2017-04-19 at 3 46 12 pm

screen shot 2017-04-19 at 3 43 52 pm

screen shot 2017-04-19 at 3 44 11 pm

screen shot 2017-04-19 at 3 44 46 pm

screen shot 2017-04-19 at 3 45 33 pm

### CODE REVIEW

File: cms/templates/cms/home_page.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
 {% wagtailuserbar %}
 {% block body_class %}template-programpage{% endblock %}
 
-{% block title %}MITx MicroMasters: Bringing MIT to you{% endblock %}
+{% block title %}MIT MicroMasters: An online MIT credential available to anyone in the world{% endblock %}
```
Comment (pdpinch) [inline cms/templates/cms/home_page.html]: Revert this change. The home page title is good as is.

File: cms/templates/cms/program_page.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
 {% load wagtailcore_tags %}
 {% load wagtailimages_tags %}
 
-{% block title %}{{ page.title }} MicroMasters{% endblock %}
+{% block title %}{{ page.title }}{% endblock %}
```
Comment (pdpinch) [inline cms/templates/cms/program_page.html]: This is also OK as is. It's useful to append 'MicroMasters' to the program name.

File: static/js/containers/LearnerPage.js [hunk 3]
```diff
@@ -39,6 +39,7 @@ class LearnerPage extends React.Component<*, LearnerPageProps, *> {
     const { params: { username }, fetchProfile } = this.props;
     fetchProfile(username);
     this.fetchDashboard();
+    document.title = "Learner's Profile";
```
Comment (pdpinch) [inline static/js/containers/LearnerPage.js]: Is it possible to include the Learner's Full name here? i.e. `[Preferred Name] [Last Name]'s MicroMasters Profile`

File: static/js/containers/ProfilePage.js [hunk 4]
```diff
@@ -24,6 +24,7 @@ class ProfilePage extends React.Component<*, ProfileContainerProps, *> {
   componentDidMount() {
     const { scrollToError } = this.props;
     scrollToError.call(this);
+    document.title = "Personal Profile";
```
Comment (pdpinch) [inline static/js/containers/ProfilePage.js]: How is this different from LearnerPage.js above?

File: ui/templates/terms_of_service.html [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {% extends "base.html" %}
 {% load i18n static %}
 
-{% block title %}{% trans "MITx MicroMasters: Terms of Service" %}{% endblock %}
+{% block title %}{% trans "Terms of Service" %}{% endblock %}
```
Comment (pdpinch) [inline ui/templates/terms_of_service.html]: Let's leave this as is.

File: static/js/containers/LearnerPage.js (hunk 3 above)
Comment (amir-qayyum-khan) [inline static/js/containers/LearnerPage.js]: can i use `micromasters/static/js/util/util.js` line  ``getUserDisplayName``

File: static/js/containers/ProfilePage.js (hunk 4 above)
Comment (aliceriot) [inline static/js/containers/ProfilePage.js]: this is the container component for all of the profile signup flow pages

Comment (amir-qayyum-khan) [inline static/js/containers/ProfilePage.js]: this is title for 3 profile tabs, personal, eduction, work, each tab has its own title

**Comment by amir-qayyum-khan** [comment]
Done @aliceriot

File: static/js/containers/LearnerPage.js [hunk 6]
```diff
@@ -95,11 +97,16 @@ class LearnerPage extends React.Component<*, LearnerPageProps, *> {
...
       };
       toRender = childrenWithProps(children, props);
+      name = `${profile.profile.preferred_name} ${profile.profile.last_name}'s`;
     }
     const { errorInfo } = profile;
-    return <Loader loaded={loaded}>
-      {errorInfo && loaded ? <ErrorMessage errorInfo={errorInfo} /> : toRender }
-    </Loader>;
+    return (
+      <DocumentTitle title={`${name} MicroMasters Profile`}>
```
Comment (aliceriot) [inline static/js/containers/LearnerPage.js]: If I do a full refresh on this page (`/learner`) I see `undefined undefined MicroMasters Profile` for a second while the request for the profile data is in-flight.

I think we should take a different approach - maybe we should have a little helper function that formats the title, something like:

```js
getDocumentTitle = () => {
  const {
    params: { username },
    profiles,
  } = this.props;
  
  if ( profiles[username] !== undefined ) {
    let first = R.propOr('', 'preferred_name', profiles[username]);
    let last = R.propOr('', 'last_name', profiles[username]);
    return `${first} ${last} MicroMasters Profile`.trim();
  }
  return 'MicroMasters Profile';
}
```

probably there's a more elegant way to express that but you get the idea. We don't ever want to stick `undefined` into a template, because `${undefined}` evaluates to the string `"undefined"`.

File: package.json [hunk 7]
```diff
@@ -79,6 +79,7 @@
     "react-bootstrap": "^0.30.8",
     "react-cropper": "^0.10.1",
     "react-datepicker": "^0.41.1",
+    "react-document-title": "^2.0.3",
```
Comment (aliceriot) [inline package.json]: oh, looks like you just did! carry on.

Comment (amir-qayyum-khan) [inline package.json]: Done @aliceriot

**Comment by pdpinch** [comment]
We're missing the site name from all these titles. I'm going to update the table on the issue and hopefully make it clearer.

**Comment by pdpinch** [comment]
Ok, I updated the description with how the page titles should look. Basically, any of the ones that we're changing in this PR should end with a pipe and the site same, i.e. `| MITx MicroMasters` -- just for branding. 

@amir-qayyum-khan if you have any questions about the list on the issue, please ask.

**Comment by amir-qayyum-khan** [comment]
Done @pdpinch @aliceriot

File: static/js/containers/LearnerPage.js [hunk 8]
```diff
@@ -72,6 +73,20 @@ class LearnerPage extends React.Component<*, LearnerPageProps, *> {
...
+  getDocumentTitle = () => {
+    const {
+      params: { username },
+      profiles,
+    } = this.props;
+
+    if (profiles[username] !== undefined) {
+      let first = R.propOr('', 'preferred_name', profiles[username]);
+      let last = R.propOr('', 'last_name', profiles[username]);
+      return `${first} ${last} | MicroMasters Profile`.trim();
```
Comment (aliceriot) [inline static/js/containers/LearnerPage.js]: did you test this to see if it works? it does not, because `profiles[username]` is an object with a key called `profile` that points to the actual profile. You'll need to do this instead:

```js
R.pathOr('', ['profile', 'preferred_name'], profiles[username]);
```

maybe a more elegant approach overall would be to do this:

```js
getDocumentTitle = () => {
  const {
    params: { username },
    profiles,
  } = this.props;
  let profilePath = [username, 'profile'];

  let first = R.pathOr('', profilePath.concat('preferred_name'), profiles);
  let last = R.pathOr('', profilePath.concat('last_name'), profiles);

  return `${first} ${last} | MITx MicroMasters Profile`
    .trim()
    .replace(/^\|\s/, '');
};
```

the current solution can, if `first` and `last` end up being `""`, render the page title as `| MicroMasters Profile`.

**Comment by amir-qayyum-khan** [comment]
Done @aliceriot
